pyccel/codegen/printing/ccode.py

```diff
@@ -381,7 +397,7 @@ def _print_PyccelMod(self, expr):
         second = self._print(expr.args[1])
 
         if expr.dtype is NativeInteger():
-            return "{} % {}".format(first, second)
+            return "(({}) % {})".format(first, second)
```

Comment:
Do you need all these parentheses? Does pyccel not add enough?
Do you have an example

Reply:
if we have a python code as follows:
```
@allow_negative_index('a')
def array_negative_indexing():
      from numpy import array
      a = array([1,2,3])
      i = -1
      a[i] = a[i - 1]
```
the last line in C is translated to 
`a.nd_long[get_index(a, i % 3)] = a.nd_long[get_index(a, i - 1 % 3)];`

Reply:
Normally the parentheses are added in the syntactic stage. You're solution leads to the following code:
```
a = (4-3)%2
```
being translated to:
```
a = (((4 - 3)) % 2);
```
which means there are too many parentheses and it's not particularly readable

I have had a look at where the parentheses are added and it is in the constructor of `PyccelOperator`:
https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L342
Could you try moving the call to `handle_precedence` out of the `if` and see what happens for your case and the case above. This is linked to issue #410

Reply:
sure i will try that

Reply:
It seems to work for me. A similar operation should also be carried out in several other classes. But maybe we should leave this to be fixed with issue #410?